Packs/GCP-IAM/Integrations/GCPIAM/GCPIAM.py

```diff
@@ -0,0 +1,3715 @@
+    permissions = argToList(args.get('permissions'))
+
+    response = client.gcp_iam_project_iam_test_permission_request(project_name, permissions)
+
+    return generate_test_permission_command_output(response, readable_header=f'Project {project_name} permissions:')
+
+
+def gcp_iam_project_iam_member_add_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """
+    Add members to project policy.
+    Args:
+        client (Client): GCP API client.
+        args (dict): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+
+    """
+    project_name = args.get('project_name')
+    role = args.get('role')
+    members = argToList(args.get('members'))
+
+    iam_policy = client.gcp_iam_project_iam_policy_get_request(project_name).get("bindings", [])
+    updated_policies = add_members_to_policy(role=role, iam_policy=iam_policy, members=members,
+                                             command_name='gcp-iam-project-iam-policy-create')
+
+    client.gcp_iam_project_iam_policy_set_request(project_name, updated_policies)
+
+    command_results = CommandResults(
+        readable_output=f'Role {role} updated successfully.'
+    )
+    return command_results
+
+
+def gcp_iam_project_iam_member_remove_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """
+    Remove members from project policy.
+    Args:
+        client (Client): GCP API client.
+        args (dict): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+
+    """
+    project_name = args.get('project_name')
+    role = args.get('role')
+    members = argToList(args.get('members'))
+
+    iam_policy = client.gcp_iam_project_iam_policy_get_request(project_name).get("bindings", [])
+    updated_policies = remove_members_from_policy(role=role, iam_policy=iam_policy, members=members,
+                                                  command_name='gcp-iam-project-iam-policy-create')
+
+    client.gcp_iam_project_iam_policy_set_request(project_name, updated_policies)
+
+    command_results = CommandResults(
+        readable_output=f'Role {role} updated successfully.'
+    )
+    return command_results
+
+
+def gcp_iam_project_iam_policy_set_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """
+    Sets the IAM access control policy for the specified project.
+    Args:
+        client (Client): GCP API client.
+        args (dict): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+
+    """
+    project_name = args.get('project_name')
+    policy = args.get('policy')
+    if isinstance(policy, str):
+        policy = policy.replace("\'", "\"")
+        if policy and not policy.startswith('['):
+            policy = '[' + policy + ']'
+
+        policy = json.loads(policy)
```

Comment:
Just edge case handling.
I added `policy = '[' + policy + ']'` because the API request is required list as input.

Reply:
if it's not a json string, how do you expect the policy to look like?

Reply:
it's a list of JSON elements, for example:
`[{'role': 'roles/resourcemanager.folderAdmin', 'members': ['user:yehuda@xsoar.com]}, {'role': 'roles/resourcemanager.folderEditor', 'members': ['user:yehuda@xsoar.com']}]`

When the policy is provided from XSOAR context,  it's an instance of `list` (and not `str`), and this case is fine.
For example:
!gcp-iam-project-iam-policy-set project_name=${GCPIAM.Project.name} policy=${GCPIAM.Policy.bindings}

But when the input is provided by the user, it may miss or input wrong characters in the policy value, so I tried to handle cases where the user provided the string in certain string forms.
For example:
!gcp-iam-folder-iam-policy-set folder_name="folders/123" policy="{\"role\": \"roles/resourcemanager.folderAdmin\",\"members\": [\"user:yehuda@xsoar.com\"]},{\"role\": \"roles/resourcemanager.folderEditor\",\"members\": [\"user:yehuda@xsoar.com\"]}"